docs/src/rfc/text/0000-core-static.md

```diff
@@ -0,0 +1,307 @@
+       e.g. `type Extractor: SectionExtractor`
+    2. The `Platform` trait uses static trait functions to specify platform configurations that is not "const init-able"
+       such as something that requires an allocation
+    3. `const` values for setting configurations that can be created at compile-time
+    4. The Platform trait itself will remain stateless.
+
+## Unresolved Questions
+
+- How to handle components like the advanced logger that need the hob list. Maybe provide it in the `components` function?
+
+## Prior Art (Existing PI C Implementation)
+
+Previous to this RFC, static state is spread across the entire repository, and platforms currently registered configuration
+for the core via dynamic polymorphism.
+
+## Alternatives
+
+N/A, keep the same.
+
+## Rust Code Design
+
+### Addition of the `Platform` trait
+
+The first design change is that we will introduce a `Platform` trait that will fully describe all platform
+customizations available. This includes both function implementations for configuration, and type specifications for
+generic trait resolution. If you remember previously, we did this in the core itself, and it lead to a laundry list of
+traits with complex types specified. This interface is cleaner, and provides less complexities in the `Core`.
+
+To start, this trait will have a limited number of configuration options as shown below, but it is expected that more
+will be specified as needed. Note that this trait is now what adds components / configs / services, instead of the core.
+
+```rust
+// In patina_dxe_core/src/lib.rs
+trait Platform {
+    const PRIORITIZE_32_BIT_MEMORY: bool = false;
+    type Extractor: SectionExtractor;
+
+    fn section_extractor() -> Self::SectionExtractor;
+
+    fn components(add: &mut Add<Component>) { }
+
+    fn configs(add: &mut Add<Config>) {  }
+
+    fn services(add: &mut Add<Service>) {  }
+
+    #[cfg(target_arch = "aarch64")]
+    fn gic_bases() -> GicBases;
+}
+```
+
+This will introduce changes to the core and how it consumes the trait. An example is shown below
+
+```rust
+pub struct Core<P: Platform> {
+    ...
+    _p: PhantomData<P>
+}
+
+impl <P: Platform> Core<P> {
+    /// This was originally `init_memory`. It is now `start` and the only public function.
+    pub fn start(mut self, physical_hob_list: *const c_void) -> ! {
+        ...
+
+        GCD.prioritize_32_bit_memory(P::PRIORITIZE_32_BIT_MEMORY);
+
+        ...
+    }
+
+    // This was originally `start`. It is now a method (probably not dispatch) called by `start` above.
+    fn dispatch(&mut self) -> Result<()> {
+        ...
+
+        Self::register_section_extractor(P::section_extractor());
+
+        ...
+    }
+}
+```
+
+### Creation of the `UefiState` struct
+
+Since we will be moving all of the static state into the Core, we need a clean way to store all of this state. To do
+that, we will create a new struct called `UefiState` that stores everything that is currently static. Please note that

```

Comment:
What is the benefit of putting them in a sub-struct instead of just directly into the core?

Reply:
Mainly it was a personal preference to separate out the data. At a rough count I saw about 20 statics spread across `patina_dxe_core` and I did not exactly want that many fields in the top-level Core struct.

It also slightly separates out the "uefi" functionality from the "component" functionality, which I like.

Reply:
We could theoretically move all the UEFI-esque functions to this struct instead of implemented on the `Core`.